### ISSUE TITLE
QNN: wav2letter doesn't export

### ISSUE THREAD

**Issue by guangy10** [pytorch/executorch#7634]
### 🐛 Describe the bug

```
python -m examples.qualcomm.scripts.wav2letter -b cmake-out -m SM8450 --compile_only -p ./states_fused.pth
```
fails to export

where states_fused.pth is downloaded from https://github.com/nipponjo/wav2letter-ctc-pytorch/tree/main?tab=readme-ov-file#wav2letter-ctc-pytorch

cc: @cccclai @cbilgin 

### Versions

latest main

**Comment by cccclai** [comment]
Is there any error message?

**Comment by winskuo-quic** [comment]
Hi @guangy10,
I am trying to reproduce this error, but everything is working fine on my side.
I am currently using `torchaudio 2.6.0.dev20250131+cpu`.

**Comment by guangy10** [comment]
@winskuo-quic If you think the model is working fine, should we just enable it in the CI, the setup there can be the source of truth for future reference. Here is the QNN models we are currently running on CI: https://github.com/pytorch/executorch/blob/main/.github/workflows/trunk.yml#L305-L329

Can you add wav2letter to it? The CI only need to test it in `--compile_only` mode, to ensure there is no complication issue

**Comment by winskuo-quic** [comment]
@guangy10 Could you please share where I can place my model's weight file? I can't use wget to retrieve the weights, so I think I might need a location to place the weights where CI can also have access to.

**Comment by guangy10** [comment]
> [@guangy10](https://github.com/guangy10) Could you please share where I can place my model's weight file? I can't use `wget` to retrieve the weights, so I think I might need a location to place the weights where CI can also have access to.

Can we use random weights instead, for compile_only, since the model is not calibrated nor evaluating the accuracy. In the CI the goal is to just ensure it's functioning end-to-end with low latency. Users can always rerun these models locally by disabling the compile_only so that they will get the high quality model easily. wdyt @winskuo-quic

**Comment by cccclai** [comment]
> random weights

Random weights possibly have a different latency number than the actual weights. If it will be used on dashboard, it's probably better to use real weights

**Comment by guangy10** [comment]
> > random weights
> 
> Random weights possibly have a different latency number than the actual weights. If it will be used on dashboard, it's probably better to use real weights

If the weights are public, one solution is to cache the weights in the docker image for qnn. It can avoid redownloading the weights again and again. And we already have that docker.

**Comment by cccclai** [comment]
> [@cccclai](https://github.com/cccclai) Curious how different the numbers are—can you share them? Or are you assuming a hypothetical scenario?

I've observed a 10-15% perf difference for a model we plan to ship to production. It took me a while to debug the perf gap and it turns out that the difference is random weight/real weight

**Comment by winskuo-quic** [comment]
@guangy10, @cccclai,
For wav2letter, we are using public weights, however, for mobilebert, we are training our own weights in the script.
For mobilebert, I would like to know if we can save our own trained weights in the docker?
Also, could you share how I can save the weights in the docker image? Thanks

**Comment by cccclai** [comment]
> how I can save the weights in the docker image

cc @kirklandsign and @huydhn for this question
